pkg/cli/node.go

```diff
@@ -467,12 +466,28 @@ func runDecommissionNodeImpl(
 		}
 
 		if !anyActive && replicaCount == 0 {
-			// We now mark the nodes as fully decommissioned.
-			req := &serverpb.DecommissionRequest{
+			// We now drain the nodes in order to close all client connections.
+			for _, targetNode := range nodeIDs {
+				drainReq := &serverpb.DrainRequest{
+					Shutdown: false,
+					DoDrain:  true,
+					NodeId:   targetNode.String(),
+				}
+				if _, err = c.Drain(ctx, drainReq); err != nil {
+					fmt.Fprintln(stderr)
+					return errors.Wrapf(err, "while trying to drain %s", targetNode.String())
+				}
+			}
+
+			// Allow time for the drains to close the client connections.
+			time.Sleep(200 * time.Millisecond)
+
+			// Finally, mark the nodes as fully decommissioned.
+			decommissionReq := &serverpb.DecommissionRequest{
```

Comment:
Are there issues if the CLI is issuing this RPC against the node that was just drained? Or does the drain process only expunge SQL connections?